File: Palindrome.py
# Given a number n, check whether it is a palindrome or not.
n=input("enter a number: ")
l1=list(n)
l2=[]
for i in range(len(l1)-1,-1,-1):
    l2.append(l1[i])
if(l2==l1):
    print("the number is a palindrome")
else:
    print("the number is not a palindrome")

File: Palindrome.cpp
// Given a number n, check whether it is a palindrome or not.
#include <iostream>
using namespace::std;

int count(int n)
{
  int i;
  for(i=0; n/10!=0;i++)
  return i;
}
    
int main()
{
  int n,n1,p,x[50], flag=0;
  cout<<"enter number you want to check:";
  cin>>n;
  n1=n;
  p = count(n);
  for(int i=1;i<=p;i++)
  {
    x[p-i]=n%10;
    n=n/10;
  }
  for(int i=0;i<p/2;i++)
  {
    if(x[i]!=x[p-i])
    {
      cout<<"number is not a palindrome";
      flag=1;
      break;
    }
    else
      continue;
  }
  if(flag==0)
    cout<<"number is a palindrome";
  return 0;
}
    
  
  
  
